### ISSUE TITLE
A non-informative error when HPC is out of storage. 

### ISSUE THREAD

**Issue by bakerhassan** [nipreps/fmriprep#2329]
I noticed that when running subjects in parallel on an HPC and you exceed the storage limit, it throws this exception below. I also noticed that the working directory is not being deleted and it consume too many Gigs and I have to manually delete it. So I'm leaving this for future enhancements. 

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/concurrent/futures/process.py", line 232, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/multiproc.py", line 70, in run_node
    result["result"] = node.result
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 217, in result
    op.join(self.output_dir(), "result_%s.pklz" % self.name)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/utils.py", line 291, in load_resultfile
    raise FileNotFoundError(results_file)
FileNotFoundError: .../fmriprep_wf/single_subject_10_wf/func_preproc_ses_2_task_rest_acq_prefrontal_wf/bold_std_trans_wf/bold_reference_wf/enhance_and_skullstrip_bold_wf/_std_target_MNI152NLin2009cAsym.resnative/skullstrip_first_dilate/result_skullstrip_first_dilate.pklz
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/concurrent/futures/_base.py", line 324, in _invoke_callbacks
    callback(self)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/multiproc.py", line 159, in _async_callback
    result = args.result()
  File "/usr/local/miniconda/lib/python3.7/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/usr/local/miniconda/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
FileNotFoundError: .../fmriprep_wf/single_subject_10_wf/func_preproc_ses_2_task_rest_acq_prefrontal_wf/bold_std_trans_wf/bold_reference_wf/enhance_and_skullstrip_bold_wf/_std_target_MNI152NLin2009cAsym.resnative/skullstrip_first_dilate/result_skullstrip_first_dilate.pklz
```

**Comment by effigies** [comment]
> I noticed that when running subjects in parallel on an HPC and you exceed the storage limit, it throws this exception below.

That's going to be a difficult thing to catch, if the HPC does not generate an `OSError` when a file is written but simply does not write the file. If you have a strategy for this, let's [open an issue on Nipype](https://github.com/nipy/nipype/issues), but I can't think of one.

> I also noticed that the working directory is not being deleted and it consume too many Gigs and I have to manually delete it.

The `--clean-workdir` can be used. Otherwise, it is intentional not to empty the working directory.

----

Closing as wontfix since this isn't something we are likely to resolve in fMRIPrep. If you have an approach in mind and are interested in proposing a PR, let me know and we can reopen.
